Legacy/bonej/src/main/java/org/bonej/plugins/SliceGeometry.java

```diff
@@ -551,20 +476,24 @@ private double calculateCentroids(final ImagePlus imp, final double min,
 					final double pixel = ip.get(x, y);
 					if (pixel >= min && pixel <= max) {
 						count++;
-						sumX += x;
-						sumY += y;
-						final double wP = pixel * m + c;
+						final double areaFraction = doPartialVolume ? filledFraction(pixel) : 1;
+						sumAreaFractions += areaFraction;
+						sumX += areaFraction * x;
+						sumY += areaFraction * y;
+						final double wP = pixel * this.m + this.c;
 						sumD += wP;
 						wSumX += x * wP;
 						wSumY += y * wP;
 					}
 				}
 			}
 			cslice[s] = count;
-			cortArea[s] = count * pixelArea;
 			if (count > 0) {
-				sliceCentroids[0][s] = sumX * vW / count;
-				sliceCentroids[1][s] = sumY * vH / count;
+				// if !doPatialVolume then sumAreaFractions = count
```

Comment:
Typo `doPartialVolume`

Reply:
Comment is redundant, and the intent is handled by 
`final double areaFraction = doPartialVolume ? filledFraction(pixel) : 1;`